server/routes/snapshotListener.ts

```diff
@@ -0,0 +1,97 @@
+import { Request, Response, NextFunction } from 'express';
+import { USERS } from 'construct-ui/lib/esm/components/icon/generated/IconNames';
+import { Test } from 'mocha';
+import { MixpanelSnapshotEvents } from 'shared/analytics/types';
+import { DB } from '../database';
+import { success } from '../types';
+import SnapshotSpaceCache from '../util/snapshotSpaceCache';
+import { mixpanelTrack } from '../util/mixpanelUtil';
+import { NotificationCategories } from '../../shared/types';
+
+
+const snapshotListener = async (
+  models: DB, cache: SnapshotSpaceCache, req: Request, res: Response, next: NextFunction
+  ) => {
+
```

Comment:
We should perform request validation prior to hitting the cache, as a security measure -- remember that anyone can ping this route with any data they want, so we need to ensure that the provided request data is legitimate/valid before taking action. We may even want to check the origin, if possible.